src/components_utils/table_format.jl

```diff
@@ -0,0 +1,336 @@
+            f.specifier[:width] = value
+        end
+    end
+
+    function precision!(f::Format, value)
+        if isnothing(value)
+            f.specifier[:precision] = ""
+        else
+            check_int_value(value)
+            f.specifier[:precision] = ".$value"
+        end
+    end
+
+    function scheme!(f::Format, value)
+        check_named_value(Scheme, value)
+        f.specifier[:type] = value.value
+    end
+
+    function sign!(f::Format, value)
+        check_named_value(Sign, value)
+        f.specifier[:sign] = value.value
+    end
+
+    function symbol!(f::Format, value)
+        check_named_value(DSymbol, value)
+        f.specifier[:symbol] = value.value
+    end
+
+    function trim!(f::Format, value)
+        if value isa Bool
+            value = value ? Trim.yes : Trim.no
+        end
+        check_named_value(Trim, value)
+        f.specifier[:trim] = value.value
+    end
+
+    # Locale
+    function symbol_prefix!(f::Format, value::AbstractString)
+        if !haskey(f.locale, :symbol)
+            f.locale[:symbol] = [value, ""]
+        else
+            f.locale[:symbol][1] = value
+        end
+    end
+
+    function symbol_suffix!(f::Format, value::AbstractString)
+        if !haskey(f.locale, :symbol)
+            f.locale[:symbol] = ["", value]
+        else
+            f.locale[:symbol][2] = value
+        end
+    end
+
+    function decimal_delimiter!(f::Format, value)
+        v = check_char_value(value)
+        f.locale[:decimal] = v
+    end
+
+    function group_delimiter!(f::Format, value)
+        v = check_char_value(value)
+        f.locale[:group] = v
+    end
+
+    function groups!(f::Format, value::Union{Vector{<:Integer}, <:Integer})
+        groups = value isa Integer ? [value] : value
+        isempty(groups) && throw(ArgumentError("groups cannot be empty"))
+
+        for g in groups
+            g < 0 && throw(ArgumentError("group entry must be non-negative integer"))
+        end
+        f.locale[:grouping] = groups
+    end
+
+    # Nully
+    function nully!(f::Format, value)
+        f.nully = value
+    end
+
+    # Prefix
+    function si_prefix!(f::Format, value)
+        check_named_value(Prefix, value)
+        f.prefix = value.value
+    end
+
+    JSON3.StructTypes.StructType(::Type{Format}) = JSON3.RawType()
+
+    function JSON3.rawbytes(f::Format)
+        aligned = f.specifier[:align] != Align.default.value
+        fill = aligned ? f.specifier[:fill] : ""
+        spec_io = IOBuffer()
+        print(spec_io,
+            aligned ? f.specifier[:fill] : "",
+            f.specifier[:align],
+            f.specifier[:sign],
+            f.specifier[:symbol],
+            f.specifier[:padding],
+            f.specifier[:width],
+            f.specifier[:group],
+            f.specifier[:precision],
+            f.specifier[:trim],
+            f.specifier[:type]
+        )
+        return JSON3.write(
+            (
+                locale = deepcopy(f.locale),
```

Comment:
curious why `deepcopy` is needed here?

Reply:
You are absolutely right, it is superfluous here. Removed in cc7ebaf